### PR TITLE
RCHAIN-3109: Remove E from ISpace

### PR DESCRIPTION
**Pull request by JosDenmark**
## Overview
Since `ISpace` is officially pure, we can move error handling into the abstract effect stack `F`. This PR removes the `Either[E, ...]` wrapper from `ISpace`, removes the `E` from all store instances.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3109

### CODE REVIEW

File: rspace/src/main/scala/coop/rchain/rspace/RSpace.scala [hunk 1]
```diff
@@ -2,27 +2,25 @@ package coop.rchain.rspace
...
 import coop.rchain.rspace.internal._
 import coop.rchain.rspace.trace._
+import coop.rchain.shared.Log
 import coop.rchain.shared.SyncVarOps._
-import com.typesafe.scalalogging.Logger
-import coop.rchain.metrics.Metrics
 import org.lmdbjava.Txn
 import scodec.Codec
 
-object A {
```
Comment (lukasz-golebiewski) [inline rspace/src/main/scala/coop/rchain/rspace/RSpace.scala]: Where did this come from???

Comment (JosDenmark) [reply rspace/src/main/scala/coop/rchain/rspace/RSpace.scala]: I think @dzajkowski was feeling rebellious when he last modified RSpace.

**Comment by JosDenmark** [comment]
bors try

File: rspace/src/main/scala/coop/rchain/rspace/util/package.scala [hunk 2]
```diff
@@ -18,7 +18,7 @@ package object util {
   ): Either[E, Option[(K, Seq[R], Int)]] =
     v.map(unpackOption)
 
-  implicit def unpackEither[F[_], C, P, E, K, R](
+  implicit def unpackEitherF[F[_], C, P, E, K, R](
```
Comment (dzajkowski) [inline rspace/src/main/scala/coop/rchain/rspace/util/package.scala]: 👍

File: rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala [hunk 3]
```diff
@@ -96,7 +96,7 @@ class ReplayRSpace[F[_], C, P, E, A, R, K](store: IStore[F, C, P, A, K], branch:
     def storeWaitingContinuation(
         consumeRef: Consume,
         maybeCommRef: Option[COMM]
-    ): F[None.type] =
+    ): F[MaybeConsumeResult] =
```
Comment (dzajkowski) [inline rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala]: 👍

File: rspace/src/main/scala/coop/rchain/rspace/RSpace.scala (hunk 1 above)
Comment (dzajkowski) [reply rspace/src/main/scala/coop/rchain/rspace/RSpace.scala]: yep, my bad.

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala [hunk 4]
```diff
@@ -51,9 +51,9 @@ class Runtime[F[_]: Sync] private (
 
 object Runtime {
 
-  type RhoISpace[F[_]]       = TCPARK[F, ISpace]
-  type RhoPureSpace[F[_]]    = TCPARK[F, PureRSpace]
-  type RhoReplayISpace[F[_]] = TCPARK[F, IReplaySpace]
+  type RhoISpace[F[_]]       = TCPARK0[F, ISpace]
+  type RhoPureSpace[F[_]]    = TCPARK0[F, PureRSpace]
```
Comment (dzajkowski) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: can we kill pure r space? it's only confusing....
not for this pr

Comment (JosDenmark) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: Yep. Next PR.

**Pull request merged**